module/VuFind/src/VuFind/Recommend/SideFacets.php

```diff
@@ -493,4 +502,49 @@ public function getHierarchicalFacetSortOptions()
     {
         return $this->hierarchicalFacetSortOptions;
     }
+
+    /**
+     * Get the result count for a checkbox facet
+     *
+     * @param string $facet Facet
+     *
+     * @return ?int
+     */
+    public function getCheckboxFacetCount(string $facet): ?int
+    {
+        $checkboxFacets = $this->results->getParams()->getCheckboxFacets();
+        $delimitedFacets = $this->results->getParams()->getOptions()->getDelimitedFacets(true);
+        foreach ($checkboxFacets as $checkboxFacet) {
+            if ($facet !== $checkboxFacet['filter']) {
+                continue;
+            }
+            [$field, $value] = explode(':', $facet, 2);
+            $checkboxResults = $this->results->getFacetList([$field => $value]);
+            if (!isset($checkboxResults[$field]['list'])) {
+                return null;
+            }
+            $count = 0;
+            $truncate = substr($value, -1) === '*';
+            if ($truncate) {
+                $value = substr($value, 0, -1);
+            }
+            foreach ($checkboxResults[$field]['list'] as $item) {
+                $itemValue = $item['value'];
+                if ($delimiter = $delimitedFacets[$field] ?? '') {
+                    [$itemValue] = explode($delimiter, $itemValue);
+                }
+                if (
+                    $itemValue == $value
+                    || ($truncate
+                    && preg_match('/^' . $value . '/', $item['value']))
```

Comment:
Is it possible you need to `preg_quote` $value here?

Reply:
Yes, added.